src/Java/gtPlusPlus/core/item/circuit/ItemAdvancedChip.java

```diff
@@ -133,6 +156,82 @@ public IIcon getIcon(ItemStack stack, int pass) {
     public String getUnlocalizedName(ItemStack stack) {
 		return super.getUnlocalizedName();
     }
-	
 
+	@Override
+	public boolean receive(ItemStack stack, EntityPlayerMP player, NBTTagCompound tag) {
+		int meta = tag.hasKey("meta", Constants.NBT.TAG_BYTE) ? tag.getByte("meta") : -1;
+		if (meta < 0 || meta > 24)
+			return true;
+
+		if (!player.capabilities.isCreativeMode) {
+			Pair<Integer, BiFunction<ItemStack, EntityPlayerMP, ItemStack>> toolIndex = findConfiguratorInInv(player);
+			if (toolIndex == null) return true;
+
+			ItemStack[] mainInventory = player.inventory.mainInventory;
+			mainInventory[toolIndex.getKey()] = toolIndex.getValue().apply(mainInventory[toolIndex.getKey()], player);
+		}
+		stack.setItemDamage(meta);
+
+		return true;
+	}
+
+	@Override
+	public boolean onItemUse(ItemStack stack, EntityPlayer player, World world, int x, int y, int z, int side, float xOffset, float yOffset, float zOffset) {
+		// nothing on server side or fake player
+		if (player instanceof FakePlayer || !world.isRemote) return false;
```

Comment:
Use gtPlusPlus.core.util.minecraft.PlayerUtils.isRealPlayer(EntityLivingBase) > FakePlayer instancing. (TC doesn't extend FakePlayer)

Reply:
I thought you have patches to make it a FakePlayer?

Reply:
Why would you think that?

Reply:
Weird. I recall you showing off a patch for this on discord before covid started. IIRC to fix TT fake player converting sandstone into fluorite.

Reply:
I just used PlayerUtils.isRealPlayer(EntityLivingBase). Didn't do anything beyond that.

Reply:
It appears thaumcraft itself doesn't actually use the broken FakeThaumcraftPlayer at all. It's properly using the FakePlayer acquried from FakePlayerFactory (like everyone else). Since we do fork TT maybe I'll just go patch TT instead.